packages/legacy-client/src/client/objectSets/createObjectSetOrderByStep.ts

```diff
@@ -15,12 +15,48 @@
  */
 
 import type { ObjectTypesFrom, OntologyDefinition } from "@osdk/api";
+import type {
+  ObjectSetDefinition,
+  OrderByClause,
+} from "../../ontology-runtime";
+import type { ClientContext } from "../../ontology-runtime/ontologyProvider/calls/ClientContext";
 import type { ObjectSetOrderByStep } from "../interfaces";
 import type { OsdkLegacyObjectFrom } from "../OsdkObject";
+import { createObjectSetTerminalLoadStep } from "./createObjectSetTerminalLoadStep";
+import { mapPropertiesToOrderBy } from "./mapPropertiesToOrderBy";
 
 export function createObjectSetOrderByStep<
   O extends OntologyDefinition<any>,
   K extends ObjectTypesFrom<O>,
->(): ObjectSetOrderByStep<OsdkLegacyObjectFrom<O, K>> {
-  return {} as any;
+>(
+  clientContext: ClientContext,
+  apiName: K,
+  objectSet: ObjectSetDefinition,
+  ontologyDefinition: O,
+  orderByClauses: OrderByClause[] = [],
+): ObjectSetOrderByStep<OsdkLegacyObjectFrom<O, K>> {
+  return {
+    orderBy(predicate) {
+      const objectProperties = ontologyDefinition.objects[apiName].properties;
+      const orderBy = mapPropertiesToOrderBy<OsdkLegacyObjectFrom<O, K>>(
+        objectProperties,
+      );
+      const orderByClause = predicate(orderBy);
+
+      return createObjectSetOrderByStep(
+        clientContext,
+        apiName,
+        objectSet,
+        ontologyDefinition,
+        [...orderByClauses, orderByClause],
+      );
+    },
+    ...createObjectSetTerminalLoadStep(
```

Comment:
This is also added in OsdkObjectSet, did you mean to have it here also?

Reply:
Yeah I needed a cleaner way of doing this, addressed!